### DISCUSSION TITLE
Journaling/keeping a diary [help wanted]

### DISCUSSION THREAD

**Discussion by kslstn** [standardnotes/forum#2700] [Suggestions]
I’m using the Day One journaling app. Because most of the features I use are in Standard Notes, I’d like to do journaling in SN, making things simpler for myself. 

What I’d need for that is almost offered by SN:
✅ Separate journal category: solved with tags
- Hide entries from ‘All notes’. This is possible with archived notes, but these don’t show up under the tag. Protected notes sort of allow this too, as in that they don’t show the content in ‘All notes’. Both solutions require that I’d change the note setting for each entry, which is friction that will likely cause me to journal less and the main reason I’m still using Day One. 
- I’d like to see the date for the journal entry as a fallback for the title; I don’t want to enter a title for each. A list of protected notes without a title doesn’t provide a good browsing experience. 
- Sorting of journal entries should be by date of creation (or better, the date the entry is about, see #67. I sort all my other notes by last edited and switching the default sorting every time is cumbersome. 

I use SN in almost all contexts of my life: work, learning, quick notes etc. Just like I clean up my desk when I’m done with a hobby, so I don’t get distracted at work, I’d like SN to be cleaned up when I open it after using it in another context. I don’t mind my diary’s cover visible on my bookshelf, but it shouldn’t be open on my desk. 

Evernote actually uses the metaphor of books directly. I never took the effort of understanding what that meant though. I think SN could meet my requirements without adding an additional layer in the information architecture though. Perhaps the simplest way would be user-defined  view contexts. ‘All notes’ would be one of them. The contexts define what notes are visible, based on their labels like tag, protection and archived. 

I’d appreciate any tips for how to set up journaling with the current feature set!

**Comment by atmoio** [comment]
Journaling is an area we want to offer better support on over time. Especially with a dedicated editor.

For now, you can definitely do custom view contexts: https://standardnotes.org/help/42/how-do-i-view-a-list-of-untagged-notes-and-create-other-dynamic-filters

P.S. Funnily enough, minutes before I saw this thread, I shared with @zsoltszilvai a link to your blog, as an example of a really uniquely designed blog for inspiration. I had it bookmarked. Not sure where I had previously seen your name :) Love the style!

**Comment by arcanecode** [comment]
I'll second the request for functionality focused on journaling. I keep all kinds of stuff in SN, but have used a variety of other tools for journaling. Mostly I use journaling to keep up with the work I do each day so I can refer back to it, should any of my clients ask "just what the heck did you work on this day that you are charging us for".

I've used other tools but SN is such a core tool for me it'd be nice to have my journal in there as well. Just need to be sure it doesn't get in the way of the other ways I'm using SN.

**Comment by kslstn** [comment]
> Journaling is an area we want to offer better support on over time. Especially with a dedicated editor.
> 
> For now, you can definitely do custom view contexts: https://standardnotes.org/help/42/how-do-i-view-a-list-of-untagged-notes-and-create-other-dynamic-filters

Awesome! 

I just tried creating a custom view, for all notes except those with the `journal` tag, but got stuck at this, which yields no results:
`!["Notes", "tags", ["title", "matches", "^(?!.*journal).*$"]]`
I appreciate that the option exists, but man, regex...

PS thanks for linking to my blog :)

**Comment by atmoio** [comment]
You might want to use the `includes` syntax:

> Show all notes that have tags that start with the letter b.

```
!["B-tags", "tags", "includes", ["title", "startsWith", "b"]]
```

So something like (untested):

```
!["Not Journal", "tags", "includes", ["title", "matches", "^(?!.*journal).*$"]]
```

**Comment by hlascelles** [comment]
I'd very much like a journal style too. 

In particular, I am using CherryTree, and it has a key to create "today's entry" (it happens to [be F8](https://giuspen.com/cherrytreemanual/)).

That will create a new item with todays date (Auto-nested YYYY/MMM/DD). 

I use it less like a diary, and more like a "oh heck, I am doing some dev / have found a stack trace, I need to quickly paste what I am seeing right now somewhere as I may need it in a few months time". So maybe (a bit) more like what @arcanecode is doing.

I can happily have a go at this, except I am mostly keen for this to be a hotkey. I know Standard Notes is not hotkey heavy (yet - and it is being worked on). Looking at the [building an extension](https://docs.standardnotes.org/extensions/building-an-extension/) page, I can work with the panels hierarchy, but hotkeys are not yet a first class citizen.

@mobitar I have found where there are a few [keybindings](https://github.com/standardnotes/web/blob/02ad4b8197b0b6c653493d573ef136aa4d12e843/app/assets/javascripts/views/notes/notes_view.ts#L752). Should I wait for the 2021 rewrite before trying a "keybinding based extension"?
